### ISSUE TITLE
[feat] disable asset compression by default and don't ship brotli

### ISSUE THREAD

**Issue by Legend-Master** [tauri-apps/tauri#9941]
### Describe the problem

We currently compress assets (e.g. js files) with brotli by default, and can't be disabled if you use any official plugins (they depends on tauri with all default features, so no default features no longer works like they did in v1)

Since our installers are all in compressed formats, compressing the assets to save disk space should be opt-in like upx, at least we shouldn't need to ship brotli, it's supported on all modern browsers, and I think we can avoid shipping a brotli to decompress those assets

(Chromium support started in 2016 and WebKit in 2017, this is earlier than what we support, WebView2 started in 2020, MacOS 10.15 in 2019, webkit2gtk 4.1 is even more recent, Android 7 in 2016, only thing missing might be iOS 9 which is from 2015)

### Describe the solution you'd like

Disable brotli/assets compression by default, and if compression is on, let browsers decode those assets instead of shipping a brotli and decoding them ourselves

**Comment by amrbashir** [comment]
cc @chippers since you wrote the embedded assets stuff

**Comment by chippers** [comment]
i've actually written serving compressed assets directly when detecting the proper accept-encoding header multiple times for gzip and brotli. i don't remember at this point why it was never added or if it was removed. at the time brotli would have needed to been included because not all webviews supported brotli for all the versions we were supporting (years ago). part of the reasoning for it was disk size of the binary itself, not necessarily the installer, and brotli brought nice wins for larger applications.

definitely something we can revisit now

**Comment by Drevoed** [comment]
Any news on this?

**Comment by FabianLars** [comment]
iirc the point with the plugins was solved so this should hopefully be a simple change now

**Comment by Legend-Master** [comment]
> iirc the point with the plugins was solved

Yeah, disable compression feature works now with official plugins

> and if compression is on, let browsers decode those assets instead of shipping a brotli and decoding them ourselves

I found out that you can get those assets from the rust side a while back, and I feel like this might not be doable anymore (sending to js side and getting the decompressed asset back feels wrong to me)

So the only question remaining might be should this feature be an opt-in or opt-out

**Comment by FabianLars** [comment]
We could just document how to use the brotli crate in the asset resolver docs. Considering that this API is not used very often I think that'd be good enough. Though probably something for v3.
